src/textual/_animator.py

```diff
@@ -64,6 +64,23 @@ def __call__(self, time: float) -> bool:  # pragma: no cover
         """
         raise NotImplementedError("")
 
+    async def invoke_callback(self) -> None:
+        """Calls the [`on_complete`][Animation.on_complete] callback if one is provided."""
+        if self.on_complete is not None:
+            await invoke(self.on_complete)
+
+    async def stop(self, complete: bool = True) -> None:
+        """Stop the animation.
+
+        Args:
+            complete: Flag to say if the animation should be taken to completion.
+
+        Note:
+            [`on_complete`][Animation.on_complete] will be called regardless
+            of the value provided for `complete`.
+        """
+        await self.invoke_callback()
```

Comment:
The `complete` flag doesn't appear to be used?

Reply:
Could we make this abstract?

Without it, you could create a valid but subtly broken `Animation` subclass.

i.e. rather than a subclass call super, have it call `invoke_callback`...